### ISSUE TITLE
[Feature Request] Add warning on delete folder

### ISSUE THREAD

**Issue by MauGal** [getinsomnia/insomnia#6487]
### Expected Behavior

When I delete a folder, I expect there to be a message advising that all requests will be deleted

### Actual Behavior

I click delete and the folder is deleted without any warning

### Reproduction Steps

1. Right click on folder
2. Click on delete 
Folder deleted

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Insomnia Version

2023.5.8

### What operating system are you using?

Windows

### Operating System Version

Windows 11

### Installation method

Downloaded from Insomnia

**Comment by rcfja** [comment]
Yeah I accidentally deleted some stuff too. Please make it much harder to accidentally delete requests or folders or anything.

This was not a fun experience for me.

**Comment by sunny-chung** [comment]
If I single click on the 6th or 7th row's arrow, the mouse down event triggers a context menu, and the mouse up event triggers the delete button. As a result, the click would delete the entire folder / request that stayed on that row without requiring user confirmation, and no undo is available.

This experience makes me to rethink if I should continue to use Insomnia.

**Comment by alexandrepeternela** [comment]
This problem continues!!!!! It deleted my entire project folder without me intending to, with a simple click on the "arrow" all my subfolders and requests disappeared !!!!! I am very disappointed!!!!!

Version: Insomnia 8.3.0
Build date: 13/10/2023
OS: Linux x64 6.2.0-35-generic
Electron: 25.8.1
Node: 18.15.0
Node ABI: 116
V8: 11.4.183.29-electron.0
Architecture: x64

**Comment by luizamorimsouza** [comment]
This problem is really annoying. I have already deleted the request folders at least three times

**Comment by sunny-chung** [comment]
I assure they won't fix it in this year. They marked the bug fix request as a "feature request", and ignored the PR which can fix the issue. Looks like there is a great deviation on the definition of bugs between Kong and us.

I developed an [alternative](https://github.com/sunny-chung/hello-http) primarily because of this issue. The alternative ports many of the existing Insomnia features.
